### ISSUE TITLE
Error while saving object using DbSession and PostgreSQL

### ISSUE THREAD

**Issue by klimov-paul** [yiisoft/yii2#11498]
### What steps will reproduce the problem?

Attempt to save object into a session variable:

```php
Yii::$app->session->set('user', Yii::$app->assetManager);
Yii::$app->session->close(); // throws an exception
exit;
```

Session table DDL:

```sql
create table "session"
(
    "id"  varchar(256) not null,
    "expire" integer,
    "data" bytea,  /* Binary string column has been used as it should be, text - would not work */
    primary key ("id")
);
```

### What is the expected result?

Object stored into the session successfully and available for the get.

### What do you get instead?

Database exception:

```
Exception (Database Exception) 'yii\db\Exception' with message 'SQLSTATE[22P02]: Invalid text representation: 7 ERROR: invalid input syntax for type bytea The SQL being executed was: UPDATE "session" SET "data"='__flash|a:0:{}_language|s:2:"ru";__id|i:1;test|s:4:"some";foo|s:4:"some";user|O:20:"yii\web\AssetManager":17:{s:7:"bundles";a:1:{s:28:"yii\bootstrap\BootstrapAsset";a:2:{s:10:"sourcePath";s:28:"@app/assets/source/bootstrap";s:3:"css";a:1:{i:0;s:18:"css/bootstrap.less";}}}s:8:"basePath";s:43:"/home/pklimov/projects/soft-farm/web/assets";s:7:"baseUrl";s:25:"/pklimov/soft-farm/assets";s:8:"assetMap";a:0:{}s:10:"linkAssets";b:1;s:8:"fileMode";N;s:7:"dirMode";i:509;s:10:"beforeCopy";N;s:9:"afterCopy";N;s:9:"forceCopy";b:0;s:15:"appendTimestamp";b:1;s:12:"hashCallback";N;s:35:"', "expire"=1462357997 WHERE "id"='85gdh4kpu4d43ck9f288sal7d1'' in /home/pklimov/projects/yii2/framework/db/Schema.php:633 Stack trace: #0 /home/pklimov/projects/yii2/framework/db/Command.php(854): yii\db\Schema->convertException(Object(PDOException), 'UPDATE "session...') #1 /home/pklimov/projects/yii2/framework/web/DbSession.php(179): yii\db\Command->execute() #2 [internal function]: yii\web\DbSession->writeSession('85gdh4kpu4d43ck...', '__flash|a:0:{}_...') #3 /home/pklimov/projects/yii2/framework/web/Session.php(175): session_write_close() #4 [internal function]: yii\web\Session->close() #5 {main}
```

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.8
| PHP version      | 5.4
| Operating system | CentOS
| PostgreSQL version | 9.4.5

The problem can be fixed using direct setup of the data PDO type to be `PDO::PARAM_LOB`.
This will work for PostgreSQL. But I am unsure if it is safe to do so for other DBMS.

Also such classes as `yii\caching\DbCache` and `yii\rbac\DbManager` may be affected by this problem.

**Comment by klimov-paul** [comment]
Solution proposed at #11499

**Comment by klimov-paul** [comment]
Resolved by commit 6b607d0
